src/nxt_file.c

```diff
@@ -310,9 +310,13 @@ nxt_file_rename(nxt_file_name_t *old_name, nxt_file_name_t *new_name)
  *
  * Linux 2.4 uses BKL for ioctl() and fcntl(F_SETFL).
  * Linux 2.6 does not use BKL.
+ * 
+ * On Solaris, the FIONBIO ioctl command is not allowed on a signalfd
+ * file descriptor. Ref.: 
+ * https://github.com/illumos/illumos-gate/blob/master/usr/src/uts/common/io/signalfd.c#L623
  */
 
-#if (NXT_HAVE_FIONBIO)
+#if (NXT_HAVE_FIONBIO) && !(NXT_SOLARIS)
```

Comment:
I've just talked to Igor, and it seems that this problem is caused by the ordering in `https://github.com/nginx/unit/blob/master/src/nxt_service.c`. Because regular Solaris system doesn't support epoll. It's expected that Solaris event port will be used instead. All these problems seem specific to illumos.

Reply:
Ok, I just got a proper Solaris 11.4 box and it seems this indeed is not needed there. But the other changes apply (the byte_order needs work and also the rearrange of the sendfile detection).